pkg/systemd/quadlet/quadlet.go

```diff
@@ -672,6 +674,11 @@ func ConvertContainer(container *parser.UnitFile, names map[string]string, isUse
 		return nil, err
 	}
 
+	groupAdd, ok := container.Lookup(ContainerGroup, KeyGroupAdd)
+	if ok && len(groupAdd) > 0 {
+		podman.add("--group-add", groupAdd)
+	}
+
```

Comment:
The code you added searches only for one occurrence of `GroupAdd` and not all of them. In order for your test to pass, you need to look for all of the occurrences and add each one of them:
```suggestion
	groupsAdd, ok := container.LookupAll(ContainerGroup, KeyGroupAdd)
	for _, groupAdd := range groupsAdd {
	    if len(groupAdd) > 0 {
		    podman.add("--group-add", groupAdd)
	    }
    }
```